### PR TITLE
Add support for "get product by ID" query 

### PR DESCRIPTION
**Pull request by JanVanekRelitas**
## What does this PR do and why?

_Describe in detail what your pull request does and why._

## What is added/fixed/removed?

**Added**
- _Write what is added_

**Fixed**
- _Write what is fixed_

**Removed**
- _Write what is removed_

## Screenshot or screen recording

### CODE REVIEW

File: modules/magento/repositories/useUrlResolverRepository.ts [hunk 1]
```diff
@@ -7,15 +7,17 @@ export default () => {
   return async (
     url: string,
   ): Promise<{
-    id: string
+    entityUid: string
+    id: number
```
Comment (adamjedlicka) [inline modules/magento/repositories/useUrlResolverRepository.ts:11]: Id should stay as string (if Magento returns it as number, convert it to string). Numbers are used for values that will be used in computations (addition, subtraction, ...). If you have data that you do not expect to perform any mathematical computation on it (e.g. IDs), treating it as string, even though its actually a number is generally a better idea.

Comment (adamjedlicka) [reply modules/magento/repositories/useUrlResolverRepository.ts:11]: We should also mark the id as deprecated (Magento marked it as deprecated already)

File: modules/cart-magento/mappers/ToCartDiscount.ts [hunk 2]
```diff
@@ -2,5 +2,5 @@ import ToMoney from '#ioc/mappers/ToMoney'
 
 export default (data: any) => ({
   amount: ToMoney(data.amount),
-  label: data.label as string,
+  label: (data.label !== ' ' && (data.label as string)) || ('' as string),
```
Comment (adamjedlicka) [inline modules/cart-magento/mappers/ToCartDiscount.ts:5]: image

Comment (adamjedlicka) [reply modules/cart-magento/mappers/ToCartDiscount.ts:5]: If this is something truly non-negotiable with BE, at least use .trim() string method.

**Review by adamjedlicka — CHANGES_REQUESTED** [review]

File: modules/magento/repositories/useUrlResolverRepository.ts (hunk 1 above)
Comment (JanVanekRelitas) [reply modules/magento/repositories/useUrlResolverRepository.ts]: fixed

**Review by adamjedlicka — APPROVED** [review]

**Review by adamjedlicka — APPROVED** [review]

**Pull request merged**